### ISSUE TITLE
Touching documents in an "in parallel" section makes no sense

### ISSUE THREAD

**Issue by bzbarsky** [w3c/ServiceWorker#1108]
https://w3c.github.io/ServiceWorker/#client-navigate runs steps in parallel (step 6) that do things like:

> If browsingContext has discarded its Document, reject promise with a TypeError and abort these steps.

But you can't poke that sort of state in an "in parallel" section.  Not least because it can change while the "in parallel" section is running (that's what "in parallel" means).

Of course it can also change before the task queued in step 6 substep 7 runs...

**Comment by bzbarsky** [comment]
@wanderview

**Comment by jungkees** [comment]
Right. I think we should move this step to the task to be queued (i.e to step 6.7.1.) One check point is whether we need this guard in the first place. Doesn't [navigate](https://html.spec.whatwg.org/#navigate) algorithm validate all the pre-navigate conditions itself? @jakearchibald, could you check if you needed this for some specific reason?

**Comment by jakearchibald** [comment]
@jungkees I think you're right, we don't need that step at all.
